### ISSUE TITLE
[prepInputs] Issue when the raster is not loaded from disk but created in the R session

### ISSUE THREAD

**Issue by ygc2l** [PredictiveEcology/reproducible#101]
This doesn't work:

```
smallRT <- prepInputs(url = "https://drive.google.com/open?id=1WhL-DxrByCbzAj8A7eRx3Y1FVujtGmtN")
test <- raster(nrow = 1000, ncol = 1000, res = 250)
postProcess(x = test, rasterToMatch = smallRT, maskWithRTM = TRUE)
```

But this does:
```
smallRT <- prepInputs(url = "https://drive.google.com/open?id=1WhL-DxrByCbzAj8A7eRx3Y1FVujtGmtN")
tempfile <- tempfile(fileext = ".tif")
writeRaster(raster(nrow = 1000, ncol = 1000, res = 250), filename = tempfile)
test <- raster(tempfile)
postProcess(x = test, rasterToMatch = smallRT, maskWithRTM = TRUE)
```

**Comment by eliotmcintire** [comment]
`raster` package weirdness:
```
> test <- raster(nrow = 1000, ncol = 1000, res = 250)
> inMemory(test)
[1] FALSE
> test[] <- 1
> inMemory(test)
[1] TRUE
```
So, the test to find out if it is on disk and has a filename fails because this function (`inMemory`) returns "FALSE" when there are no data in the `RasterLayer`

**Comment by eliotmcintire** [comment]
The initial issue examples is faulty, as there is no `crs` and only 1 cell, which means that even after fixing the bug that appears, `postProcess` won't work correctly. Nevertheless, the bug related to filenames is now fixed with 98101bc2f37f79ef8d76c6f35d78e6d0fe47635e. This version of the initial question works:
```
smallRT <- prepInputs(url = "https://drive.google.com/open?id=1WhL-DxrByCbzAj8A7eRx3Y1FVujtGmtN")
    a <- raster::extent(smallRT)
    a <- raster::extend(a, -3.5e5) # make it small
    test <- raster(a, res = 250, vals = 1)
    crs(test) <- crs(smallRT)
    a <- postProcess(x = test, rasterToMatch = smallRT, maskWithRTM = TRUE)
```
